Update documentation (#40)
kyma-incubator/service-catalog: docs/design.md

![Service Catalog Design](images/sc-design.svg)

The above is the high level architecture of Service Catalog.

> Service Catalog has two basic building blocks: a webhook server and a controller.

Reviewer: ```suggestion
Service Catalog has two basic building blocks: a Webhook Server and a controller.
```

### Webhook Server

> The Webhook Server makes use of [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 

Reviewer: ```suggestion
The Webhook Server uses [Admission Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) 
```

> to manage custom resources. Admission Webhooks is a feature available in the Kubernetes API server, 

Reviewer: ```suggestion
to manage custom resources. Admission Webhook is a feature available in the Kubernetes API Server, 
```

> that allows to implement arbitrary control decisions like validation or mutation.

Reviewer: ```suggestion
that allows you to implement arbitrary control decisions, such as validation or mutation.
```

> For every resource managed by Service Catalog (the ones listed in the previous section) there is a separate handler defined. You can see the structure at 

Reviewer: ```suggestion
For every resource managed by Service Catalog (the ones listed in the previous section), there is a separate handler defined. You can see the structure at 
```

Reviewer: I do not understand the part in ( ).

[`pkg/webhook/servicecatalog`](https://github.com/kyma-incubator/service-catalog/tree/crds/pkg/webhook/servicecatalog).

> The current version of all Service Catalog API resources is `v1beta1`, they are all defined here:

Reviewer: ```suggestion
The current version of all Service Catalog API resources is `v1beta1`. The resources are defined here:
```

[`pkg/apis/servicecatalog/v1beta1/types.go`](https://github.com/kubernetes-sigs/service-catalog/blob/master/pkg/apis/servicecatalog/v1beta1/types.go).

> If you want to learn more about Admission Webhooks read [this](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) guide.

Reviewer: ```suggestion
If you want to learn more about Admission Webhooks, read [this](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) document.
```
